python/cudf/cudf/tests/test_array_ufunc.py

```diff
@@ -70,35 +70,34 @@ def test_ufunc_index(ufunc):
     try:
         got = ufunc(*args)
     except AttributeError as e:
-        # We xfail if we don't have an explicit dispatch and cupy doesn't have
-        # the method so that we can easily identify these methods. As of this
+        # We check if an attribute error is raised so that if we don't
+        # have an explicit dispatch and cupy doesn't have the method
+        # so that we can easily identify these methods. As of this
         # writing, the only missing methods are isnat and heaviside.
         if "module 'cupy' has no attribute" in str(e):
-            pytest.xfail(reason="Operation not supported by cupy")
-        raise
+            with pytest.raises(AttributeError):
+                raise e
+            return
+        else:
+            raise
 
     expect = ufunc(*(arg.to_pandas() for arg in pandas_args))
 
-    try:
+    if (
+        fname in {"bitwise_and", "bitwise_or", "bitwise_xor"}
+        and not PANDAS_GE_150
+    ):
+        # https://github.com/pandas-dev/pandas/issues/46769
+        context = pytest.raises(AssertionError)
+    else:
+        context = nullcontext()
+
+    with context:
```

Comment:
We're expecting an `AssertionError` in pandas < 1.5 rather than `xfailing` programmatically; this pattern repeats...

Reply:
switched to `requests.applymarker(pytest.mark.xfail(condition=when, reason="foo"))`.